src/concurrency/sync.rs

```diff
@@ -696,7 +696,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         retval_succ: Scalar,
         retval_timeout: Scalar,
         dest: MPlaceTy<'tcx>,
-        errno_timeout: Scalar,
+        errno_timeout: shims::io_error::IoError,
```

Comment:
```suggestion
        errno_timeout: IoError,
```
Happens again further down this file